docs/plan/Proposal.tex

```diff
@@ -120,6 +122,64 @@ \section{Proposal}
 the daemon to fail without the system being rendered unusable.
 
 \subsection{Architecture}
+There will be two executables:
+\begin{itemize}
+  \item \texttt{utlosed}
+  \item \texttt{utloseadm}
+\end{itemize}
+Where \texttt{utlosed} will be the daemon that controls the starting and maintaining of services, and
+\texttt{utloseadm} which is the frontend to be used by both users and other applications.
+The two applications will be connected via ZeroMQ\footfullcite{zmq-home} sockets. There will be request and reply sockets
+for control, but also publisher and subscribe sockets. These will be used for distributed logging and alarms.
+ZeroMQ is a good choice because of curveZMQ\footfullcite{cmq-home}. Encrypted sockets (to allow for distributed control)
+can be achieved with only minor changes to the code base.
+
+The messaging protocol is not decided yet, but it will be open. This will allow for third parties to easily
+write other frontends, or log collectors.
+
+\subsubsection{Logging and alarms}
+All output of the services will be redirected back to \texttt{utlosed} which will both write to a rotating log
+(configurable) and broadcast. The channel will be configurable, but the default will be: \texttt{logs/hostname/service}.
+This means that any logging client can filter for logs, logs from a specific host, and logs from a specific service on a specific
+host.
+
+Logs written to \texttt{stderr} (and crash reports detected by \texttt{utlosed}) will also be reported on \texttt{alarms/hostname/service}
+allowing for a service to listen specifically for things that are going wrong.
+
+All logs currently held on disk may be requested via the request responce socket interface. This allows remote clients to both
+respond live events happening (via subscription), but doesn't have to be running 100\% of the time to get a full history.
```

Comment:
1. Should probably be "respond **to** live events"?
2. Also I'm not sure about the "doesn't". It refers to the remote clients, right? Them it should probably be "but those don't have to be running".